OpenRA.Mods.Common/Traits/Render/RenderDebugState.cs

```diff
@@ -65,7 +65,7 @@ public void OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 
 		Color GetColor()
 		{
-			return self.EffectiveOwner != null ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
+			return self.EffectiveOwner != null && self.IsInWorld ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
```

Comment:
This is a pretty tough one. The stacktrace doesn't tell much except that `OnOwnerChanged` triggers `GetColor()` not even with line numbers. Somehow querying `EffectiveOwner` for actors outside the world or newly added to the world crashes badly. Maybe caching is responsible https://github.com/OpenRA/Mailaender/OpenRA/blob/infiltration-mission/OpenRA.Game/Actor.cs#L116 Will do some experiments to solve this elsewhere, but it will probably get even messier.